### PR TITLE
EVA-2875 - Create common groovy utilities

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**
EVADatabaseEnvironment - Database environment with a "bag of objects" from the EVA object model (ex: SubmittedVariantEntity, ClusteredVariantEntity, mongoTemplate etc.,)

RetryableBatchingCursor - A MongoDB query cursor that returns results in batches 

EVAObjectModelUtils - Basic utility functions to transform and shape objects in the EVA object model

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Run tests
```
Comment (sundarvenkata-EBI) [inline .github/workflows/run-tests.yml:1]: Github Action was verified locally using [act](https://github.com/nektos/act)

File: src/main/groovy/uk/ac/ebi/eva/groovy/commons/RetryableBatchingCursorIterator.groovy [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+class RetryableBatchingCursorIterator<T> implements Iterator<List<T>> {
+    MongoTemplate mongoTemplate
+    Class<T> collectionClass
+    MongoCursor<Document> mongoResultIterator
+    int pageSize
+
+    final static def logger = LoggerFactory.getLogger(RetryableBatchingCursorIterator.class)
+
+    // Need this to satisfy Spring gods who feast on empty constructors
```
Comment (tcezard) [inline src/main/groovy/uk/ac/ebi/eva/groovy/commons/RetryableBatchingCursorIterator.groovy:36]: 😄

File: src/main/groovy/uk/ac/ebi/eva/groovy/commons/RetryableBatchingCursorIterator.groovy [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import org.bson.Document
+import org.slf4j.LoggerFactory
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.retry.RetryCallback
+import org.springframework.retry.RetryContext
+import org.springframework.retry.backoff.FixedBackOffPolicy
+import org.springframework.retry.policy.SimpleRetryPolicy
+import org.springframework.retry.support.RetryTemplate
+
+class RetryableBatchingCursorIterator<T> implements Iterator<List<T>> {
```
Comment (tcezard) [inline src/main/groovy/uk/ac/ebi/eva/groovy/commons/RetryableBatchingCursorIterator.groovy:28]: If I understand correctly, this retry the fetching of results but will not protect against lost cursor on the server side.
Is that right ?

**Review by tcezard — APPROVED** [review]

**Review by nitin-ebi — APPROVED** [review]

File: src/main/groovy/uk/ac/ebi/eva/groovy/commons/EVADatabaseEnvironment.groovy [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+    /***
+     * Create a "bag of objects" pertinent to the EVA variant model
+     * @param propertiesFile Application properties file (ex: application.properties)
+     * that is used to run Spring-based pipelines
+     * @param springApplicationClass Application class that should be wired (ex: if an environment for
+     * accessioning pipeline is needed, use uk.ac.ebi.eva.depre)
+     * @return A database Environment
+     */
+    static EVADatabaseEnvironment createFromSpringContext(String propertiesFile, Class springApplicationClass) {
```
Comment (tcezard) [inline src/main/groovy/uk/ac/ebi/eva/groovy/commons/EVADatabaseEnvironment.groovy:74]: Is there any way to get those groovy environment to get created from the maven files ? 
That would avoid the proliferation of the properties files everywhere.

File: src/main/groovy/uk/ac/ebi/eva/groovy/commons/RetryableBatchingCursorIterator.groovy (hunk 3 above)
Comment (sundarvenkata-EBI) [reply src/main/groovy/uk/ac/ebi/eva/groovy/commons/RetryableBatchingCursorIterator.groovy:28]: Yes, correct

File: src/main/groovy/uk/ac/ebi/eva/groovy/commons/EVADatabaseEnvironment.groovy (hunk 4 above)
Comment (sundarvenkata-EBI) [reply src/main/groovy/uk/ac/ebi/eva/groovy/commons/EVADatabaseEnvironment.groovy:74]: You mean directly use maven-settings.xml in the configuraion repo? Application properties have more information than what is present in the settings XML. A better approach would be to use "layered" properties file where we have a "core" properties file in a standardized location (similar to how we have accession-production-datasources.properties under the eva-accession software dir in the cluster) and then use an incremental properties file with only the parameters that are being changed (ex: database being used in the DEV database for a given remediation task). It is a bit more involved but I have created [a separate ticket](https://www.ebi.ac.uk/panda/jira/browse/EVA-3109) to investigate that.

**Pull request merged**